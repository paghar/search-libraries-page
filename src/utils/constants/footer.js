export const footerItems = [   
  {value: 1, text: "Team"},
  {value: 2, text: "Terms and Conditions"},
  {value: 3, text: "Privacy Policy"},  
  {value: 4, text: "API"},  
];