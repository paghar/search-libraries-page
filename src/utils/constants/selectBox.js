export const sortItems = [   
  {value: 1, text: "Stars"},
  {value: 2, text: "Ranks"},
  {value: 3, text: "Dependents"},  
  {value: 4, text: "Latest Release"},
  {value: 5, text: "Contributions"},
  {value: 6, text: "Newest"},  
];