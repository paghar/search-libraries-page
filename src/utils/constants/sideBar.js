export const sideBarItems = [   
  {value: 1, text: "Home"},
  {value: 2, text: "Creating Packages"},
  {value: 3, text: "API"},  
  {value: 4, text: "Configuration"},  
  {value: 5, text: "Pluggable Resolvers"},  
  {value: 6, text: "Tools"},  
  {value: 7, text: "About"},  
];