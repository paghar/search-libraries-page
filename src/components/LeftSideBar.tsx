import {sideBarItems} from "@/utils/constants/sideBar";
import {ISideBarItem} from "@/utils/interface/misc";

const LeftSideBar = ()=>{  

  const items = sideBarItems?.map((item:ISideBarItem)=>{        
    return <h1 
      className="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"  
      key={item.value} 
    >
      {item.text}
    </h1>; 
  });

  return(     
    <div className="flex flex-col items-start m-4 bg-white  rounded-lg shadow-xl p-6  lg:w-1/6 lg:block md:hidden max-sm:hidden">       
      {items}
    </div>
  );
};
    
export default LeftSideBar;
    