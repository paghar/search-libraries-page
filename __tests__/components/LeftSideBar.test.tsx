import React from "react";
import {render} from "@testing-library/react";
import LeftSideBar from "../../src/components/LeftSideBar";
import {sideBarItems} from "../../src/utils/constants/sideBar";

describe("Button Component", () => {

  const {container} = render(<LeftSideBar />);
  
  it("Renders Snapshot Buttons", () => {    
    expect(container).toMatchSnapshot();
  });
  
});

describe("LeftSideBar Component", () => {
  test("renders sidebar items correctly", () => {
    const {getByText} = render(<LeftSideBar />);

    sideBarItems.forEach(item => {
      const sidebarItem = getByText(item.text);
      expect(sidebarItem).toBeInTheDocument();
    });
  });
});