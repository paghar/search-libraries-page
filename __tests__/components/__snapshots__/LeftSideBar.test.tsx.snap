// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Renders Snapshot Buttons 1`] = `
<div>
  <div
    class="flex flex-col items-start m-4 bg-white  rounded-lg shadow-xl p-6  lg:w-1/6 lg:block md:hidden max-sm:hidden"
  >
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      Home
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      Creating Packages
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      API
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      Configuration
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      Pluggable Resolvers
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      Tools
    </h1>
    <h1
      class="text-center text-base sm:text-xs font-semibold mb-4 text-gray-800"
    >
      About
    </h1>
  </div>
</div>
`;
